.buildkite/dagster-buildkite/dagster_buildkite/python_packages.py

```diff
@@ -0,0 +1,107 @@
+import glob
+import logging
+from distutils import core as distutils_core  # pylint: disable=deprecated-module
+from importlib import reload
+from pathlib import Path
+from typing import Set
+
+from dagster_buildkite.utils import get_changed_files
+from pkg_resources import Requirement, parse_requirements
+
+
+class PythonPackage:
+    def __init__(self, setup_py_path: Path):
+        self.directory = setup_py_path.parent
+
+        # run_setup stores state in a global variable. Reload the module
+        # each time we use it - otherwise we'll get the previous invocation's
+        # distribution if our setup.py doesn't implement setup() correctly
+        reload(distutils_core)
+        distribution = distutils_core.run_setup(str(setup_py_path), stop_after="init")
+
+        self._install_requires = distribution.install_requires  # type: ignore[attr-defined]
+        self._extras_require = distribution.extras_require  # type: ignore[attr-defined]
+        self.name = distribution.get_name()  # type: ignore[attr-defined]
+
+    @property
+    def install_requires(self):
+        return set(
+            requirement
+            for requirement in parse_requirements(self._install_requires)
+            if get(requirement.name)  # type: ignore[attr-defined]
+        )
+
+    @property
+    def extras_require(self):
+        extras_require = {}
+        for extra, requirements in self._extras_require.items():
+            extras_require[extra] = set(
+                requirement
+                for requirement in parse_requirements(requirements)
+                if get(requirement.name)
+            )
+        return extras_require
+
+    def __repr__(self):
+        return self.name
```

Comment:
? seems like it could be confusing to just see the name , include class too?

Reply:
Sure - I just didn't care for whatever the default repr was.